lib/thrift/protocols/binary.ex

```diff
@@ -53,10 +49,9 @@ defmodule Thrift.Protocols.Binary do
   def serialize(_, nil) do
     []
   end
-  def serialize(:bool, value) do
-    value = bool_to_int(value)
-    <<value::8-signed>>
-  end
+  def serialize(:bool, nil),   do: <<0::8-signed>>
```

Comment:
Good point @pguillory.  

Related - some thrift clients (I know the Ruby client does this) provide validation functionality - e.g., are required fields set, is a string field actually a string, etc.  It's slow but it's usually optional.  I typically enable it for testing but disable it in production.  This might be something to add a TODO for later?